sql/stripe/subscription_volume/view.sql

```diff
@@ -0,0 +1,32 @@
+CREATE OR REPLACE VIEW
+  subscription_volume
+AS
+SELECT
+  `date`,
+  products_v1.name AS product,
+  COALESCE(`date` >= DATE(trial_end), TRUE) AS paying,
+  COUNT(DISTINCT customer) AS subscription_count,
+FROM
+  stripe_derived.subscriptions_v1
+LEFT JOIN
+  stripe_derived.plans_v1
+ON
+  (subscriptions_v1.plan = plans_v1.id)
+LEFT JOIN
+  stripe_derived.products_v1
+ON
+  (plans_v1.product = products_v1.id)
+CROSS JOIN
```

Comment:
Oh wow, haven't seen this pattern before! It took me a couple min to wrap my head around. If this isn't already a common pattern, might this query be more readably expressed as the following?

(forgive the `moz-fx-data-shared-prod` everywhere - it was needed so that I could lint the query)

```WITH date_range AS (
    SELECT `date`
    FROM UNNEST(GENERATE_DATE_ARRAY(
        (SELECT DATE(MIN(start_date)) FROM moz-fx-data-shared-prod.stripe_derived.subscriptions_v1),
        CURRENT_DATE,
        INTERVAL 1 DAY
    )) AS `date`
)
SELECT
  `date`,
  products_v1.name AS product,
  COALESCE(`date` >= DATE(trial_end), TRUE) AS paying,
  COUNT(DISTINCT customer) AS subscription_count,
FROM
  date_range
LEFT JOIN
  moz-fx-data-shared-prod.stripe_derived.subscriptions_v1
ON
  (DATE(start_date) = `date`)
LEFT JOIN
  moz-fx-data-shared-prod.stripe_derived.plans_v1
ON
  (subscriptions_v1.plan = plans_v1.id)
LEFT JOIN
  moz-fx-data-shared-prod.stripe_derived.products_v1
ON
  (plans_v1.product = products_v1.id)
WHERE
  status NOT IN ("incomplete", "incomplete_expired")
GROUP BY
  `date`,
  product,
  paying
```

Reply:
Oh man I totally lost the plot writing that query, sorry

Reply:
I think I meant the following?
```
LEFT JOIN
  moz-fx-data-shared-prod.stripe_derived.subscriptions_v1
ON
  DATE(start_date) >= `date` AND DATE(ended_at) < `date`
```

Reply:
If I understood correctly, @relud pointed out over Zoom that the left join on `subscriptions_v1` is less efficient (it creates more rows that then need to be filtered out by the join condition) than his approach

Reply:
as discussed in zoom, i'm going to keep the GENERATE_DATE_ARRAY, but isolate the application of it via CTE so that it's easier to read